### PR TITLE
Likes

### PR DESCRIPTION
**Pull request by shrushti2000**
Implemented like video feature
1. Users can like a video 
2.  Users can view  all liked videos

### CODE REVIEW

File: src/pages/LikedVideosPage/LikedVideosPage.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const LikedVideosPage = () => {
+    const { likedVideos } = useData()
+   
+    return (
+        <div className='main-page-container'>
+            <Sidebar />
+            <div className='main-page flex-hz'>
+                {likedVideos.length===0 ? <div className='flex-vt'><p className='playlist-page-text'>You haven't liked any videos yet .</p><Link to="/videos" className='links'><button className='btn btn-secondary playlist-page-link'>Get started</button></Link></div>:<>{likedVideos.map(video => {
+                    return (<>
+                       <VideoCard videoItem={video}/>
```
Comment (VishakAmin) [inline src/pages/LikedVideosPage/LikedVideosPage.js]: _**key**_ is missing

File: src/Context/Context.js [hunk 2]
```diff
@@ -45,7 +46,7 @@ const Context = ({ children }) => {
         fetchData()
     }, [])
     return (
-        <StateContext.Provider value={{ categories: state.categories, videos: state.videos, showtoast: state.showtoast, sortByLatest: state.sortByLatest, selectedCategory: state.selectedCategory,playlists:state.playlists,showPlaylistModal:state.showPlaylistModal,currentVideo:currentVideo,setCurrentVideo:setCurrentVideo, watchlater:state.watchlater,dispatch }}>{children}</StateContext.Provider>
+        <StateContext.Provider value={{ categories: state.categories, videos: state.videos, showtoast: state.showtoast, sortByLatest: state.sortByLatest, selectedCategory: state.selectedCategory,playlists:state.playlists,showPlaylistModal:state.showPlaylistModal,currentVideo:currentVideo,setCurrentVideo:setCurrentVideo,watchlater:state.watchlater,likedVideos:state.likedVideos,dispatch }}>{children}</StateContext.Provider>
```
Comment (VishakAmin) [inline src/Context/Context.js:49]: You can destruct state at the top _**state**_ has been repeated quite often

File: src/Reducer/reducerFunction.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ const reducerFunction = (state, action) => {
     case 'UPDATE_PLAY_LIST':return {...state,playlists:state.playlists.map(playlist=>playlist._id===action.payload._id ? action.payload:playlist)}
     case 'UPDATE_PLAYLISTS':return{...state,playlists:action.payload}
     case 'UPDATE_WATCH_LATER':return {...state,watchlater:action.payload}
+    case 'ADD_VIDEO_TO_LIKED_VIDEOS':return {...state,likedVideos:action.payload}
```
Comment (VishakAmin) [inline src/Reducer/reducerFunction.js:13]: I see you are using camelCase from naming but few places it's not followed. Would be good if you stick to one as it won't affect the readability. Here you can see **_likedVideos_** and **_watchlater_**

**Review by VishakAmin — COMMENTED** [review]
Looks good left few comments

File: src/Context/Context.js (hunk 2 above)
Comment (govindappaarun) [inline src/Context/Context.js:49]: I also suggest to add some formatting to the code and configure to format on save

File: src/Reducer/reducerFunction.js (hunk 3 above)
Comment (govindappaarun) [inline src/Reducer/reducerFunction.js:13]: suggestion: could have been short like LIKE_VIDEO

File: src/Utility/likeVideoService.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import axios from "axios"
+export const addVideoToLikedVideos = async (token, videoItem, dispatch, navigate) => {
+    console.log(token)
```
Comment (govindappaarun) [inline src/Utility/likeVideoService.js:3]: remove console.logs, I see one at #15

File: src/Utility/likeVideoService.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+export const removeFromlikedVideos=async(token, dispatch, videoItemId, showCTAcontainer, setShowCTAcontainer,navigate)=>{
+    try{
+        const res=await axios.delete(`/api/user/likes/${videoItemId}`,{
+            headers:{
+                authorization:token
+            }
+        })
+        dispatch({ type: 'ADD_VIDEO_TO_LIKED_VIDEOS', payload: res.data.likes })
```
Comment (govindappaarun) [inline src/Utility/likeVideoService.js:31]: Suggestion: use prettier in your fav editor, it becomes hard to read for others who use formatting setup

**Review by govindappaarun — COMMENTED** [review]
Would suggest to use formatting tools before you merge, 
Readability will be a big concern and hard to read not formatted code

**Pull request merged**